components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/internal/TemplateManagerComponent.java

```diff
@@ -100,4 +129,124 @@ protected void unsetConfigurationManager(ConfigurationManager configurationManag
         }
         TemplateManagerDataHolder.getInstance().setConfigurationManager(null);
     }
+
+    @Reference(
+            name = "identity.template.handler",
+            service = ReadOnlyTemplateHandler.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetReadOnlyTemplateHandler")
+    protected void setReadOnlyTemplateHandler(ReadOnlyTemplateHandler readOnlyTemplateHandler) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Template handler " + readOnlyTemplateHandler.getClass().getName() + " is added.");
+        }
+        TemplateManagerDataHolder.getInstance().addReadOnlyTemplateHandler(readOnlyTemplateHandler);
+    }
+
+    protected void unsetReadOnlyTemplateHandler(ReadOnlyTemplateHandler readOnlyTemplateHandler) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Template handler " + readOnlyTemplateHandler.getClass().getName() + " is removed.");
+        }
+        TemplateManagerDataHolder.getInstance().removeReadOnlyTemplateHandler(readOnlyTemplateHandler);
+    }
+
+    private void loadDefaultTemplates() throws FileNotFoundException {
+
+        List<Path> paths = Arrays.asList(TemplateMgtConstants.SP_TEMPLATES_DIR_PATH, TemplateMgtConstants
+                .IDP_TEMPLATES_DIR_PATH);
+        for (Path path : paths) {
+            if (!Files.exists(path) || !Files.isDirectory(path)) {
+                log.info("No file-based templates");
+            } else {
+                try {
+                    Files.walk(path)
+                            .filter(filePath -> Files.isRegularFile(filePath) && filePath.toString().endsWith
+                                    (TemplateMgtConstants.FILE_EXT_JSON))
+                            .forEach(filePath -> {
+                                try {
```

Comment:
do we need this nested try block? since we already have a try block at line 163?